src/validation.cpp

```diff
@@ -2843,6 +2875,55 @@ bool CChainState::ConnectBlock(const CBlock& block, CValidationState& state, CBl
             cache.EraseStoredVariables(static_cast<uint32_t>(pindex->nHeight));
         }
 
+        // Migrate loan and collateral tokens to Gov vars. Do so at +1 height so that
+        // GetLastHeight() in Gov var Validate() has a height equal to the GW fork.
+        if (pindex->nHeight == chainparams.GetConsensus().GreatWorldHeight + 1) {
+
+            std::map<DCT_ID, CLoanSetLoanToken> loanTokens;
+            std::vector<CLoanSetCollateralTokenImplementation> collateralTokens;
+
+            cache.ForEachLoanToken([&](const DCT_ID& key, const CLoanSetLoanToken& loanToken) {
+                loanTokens[key] = loanToken;
+                return true;
+            });
+
+            cache.ForEachLoanCollateralToken([&](const CollateralTokenKey& key, const uint256& collTokenTx) {
+                auto collToken = cache.GetLoanCollateralToken(collTokenTx);
+                if (collToken) {
+                    collateralTokens.push_back(*collToken);
+                }
+                return true;
+            });
+
+            // Apply fixed_interval_price_id first
+            std::map<std::string, std::string> attrsFirst;
+            std::map<std::string, std::string> attrsSecond;
+
+            int loanCount = 0, collateralCount = 0;
+            for (const auto& [id, token] : loanTokens) {
+                std::string prefix{"v0/token/" + std::to_string(id.v) + '/'};
+                attrsFirst[prefix + "fixed_interval_price_id"] = token.fixedIntervalPriceId.first + '/' + token.fixedIntervalPriceId.second;
+                attrsSecond[prefix + "loan_minting_enabled"] = token.mintable ? "true" : "false";
+                attrsSecond[prefix + "loan_minting_interest"] = KeyBuilder(ValueFromAmount(token.interest).get_real());
```

Comment:
I know this is just a one-time action, but can't we use the values from the map here with the enum as key? hardcoded string keys like that are so errorprone. :/

Reply:
We could. 
```
auto attr = cache.GetAttributes();
attr->attributes[key] = value;
...
cache.SetVariable(*attr);
```

Reply:
Updated to use the map from ATTRIBUTES.